driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -126,7 +124,7 @@ public void run() {
             }
             reusedConnection = null;
             connections.add(connection);
-            connectionFutures.add(handleErrors(setKeyspaceAsync(connectionFuture, connection, keyspace), initExecutor));
```

Comment:
Yes, setting the keyspace at pool construction was an attempt to work around the blocking USE call in borrow, but as we later found out it could still happen even with a fixed pool size, like when a connection gets replaced because of stream id depletion. [The doc](http://datastax.github.io/java-driver/2.0.11/features/async/#known-limitations) mentions that full async only works with a session with no keyspace set.
I agree that JAVA-893 will be the real fix for this.